### PR TITLE
specify install ns when enable an addon on a cluster

### PR DESCRIPTION
**Pull request by xauthulei**
Signed-off-by: Leilei Hu <leileihu@cn.ibm.com>

Fix the issue #78 and refresh doc, @qiujian16 ,  would you help review this please. Thanks

### CODE REVIEW

**Comment by xauthulei** [comment]
Some of my local testing snapshots :
```
llhu@HuLeis-MacBook-Pro clusteradm % ./bin/clusteradm enable addons --helpenable addons

Usage:
  clusteradm enable addons [flags]

Examples:

# Enable addons on a cluster in speccified a namespace
./bin/clusteradm enable addons --names application-manager --ns namespace --clusters cluster1,cluster2


Flags:
      --clusters string      Names of the managed cluster to deploy the add-on to (comma separated)
  -h, --help                 help for addons
      --names string         Names of the add-on to deploy (comma separated)
      --ns string            Specified namespace to enable addon (default "open-cluster-management-agent-addon")
      --output-file string   The generated resources will be copied in the specified file

Global Flags:
      --add-dir-header                   If true, adds the file directory to the header of the log messages
      --alsologtostderr                  log to standard error as well as files
      --as string                        Username to impersonate for the operation
      --as-group stringArray             Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
      --cache-dir string                 Default cache directory (default "/Users/llhu/.kube/cache")
      --certificate-authority string     Path to a cert file for the certificate authority
      --client-certificate string        Path to a client certificate file for TLS
      --client-key string                Path to a client key file for TLS
      --cluster string                   The name of the kubeconfig cluster to use
      --context string                   The name of the kubeconfig context to use
      --dry-run                          If set the generated resources will be displayed but not applied
      --insecure-skip-tls-verify         If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure
      --kubeconfig string                Path to the kubeconfig file to use for CLI requests.
      --log-backtrace-at traceLocation   when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                   If non-empty, write log files in this directory
      --log-file string                  If non-empty, use this log file
      --log-file-max-size uint           Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is unlimited. (default 1800)
      --logtostderr                      log to standard error instead of files (default true)
      --match-server-version             Require server version to match client version
  -n, --namespace string                 If present, the namespace scope for this CLI request
      --one-output                       If true, only write logs to their native severity level (vs also writing to each lower severity level)
      --password string                  Password for basic authentication to the API server
      --request-timeout string           The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests. (default "0")
  -s, --server string                    The address and port of the Kubernetes API server
      --skip-headers                     If true, avoid header prefixes in the log messages
      --skip-log-headers                 If true, avoid headers when opening log files
      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
      --timeout int                      extend timeout from 300 secounds  (default 300)
      --tls-server-name string           Server name to use for server certificate validation. If it is not provided, the hostname used to contact the server is used
      --token string                     Bearer token for authentication to the API server
      --user string                      The name of the kubeconfig user to use
      --username string                  Username for basic authentication to the API server
  -v, --v Level                          number for the log level verbosity
      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging
```

**Verify**
```
llhu@HuLeis-MacBook-Pro clusteradm % kubectl get mcl
NAME     HUB ACCEPTED   MANAGED CLUSTER URLS      JOINED   AVAILABLE   AGE
llhu-1   true           https://127.0.0.1:50913   True     True        6m55s
llhu-2   true           https://127.0.0.1:50913   True     True        5m53s
llhu@HuLeis-MacBook-Pro clusteradm % bin/clusteradm install addons --names application-manager
Installing built-in application-manager add-on to the Hub cluster...
llhu@HuLeis-MacBook-Pro clusteradm % bin/clusteradm enable addons --names application-manager --clusters llhu-1
Deploying application-manager add-on to namespaces open-cluster-management-agent-addon of managed cluster: llhu-1.
llhu@HuLeis-MacBook-Pro clusteradm % bin/clusteradm enable addons --names application-manager --clusters llhu-1 --ns test
Deploying application-manager add-on to namespaces test of managed cluster: llhu-1.
```

File: pkg/cmd/enable/addons/exec.go [hunk 1]
```diff
@@ -117,15 +118,15 @@ func (o *Options) runWithClient(clusterClient clusterclientset.Interface,
 	for _, addon := range o.values.addons {
 		if addon == appMgrAddonName {
 			for _, clusterName := range o.values.clusters {
-				cn := &ClusterName{ClusterName: clusterName}
+				cn := &ClusterName{ClusterName: clusterName, NameSpace: o.namespace}
```
Comment (kim-fitness) [inline pkg/cmd/enable/addons/exec.go:121]: I think we should give the name space a default value `open-cluster-management-agent-addon`

**Comment by qiujian16** [comment]
@xauthulei it looks fine but test fails..

/approve

**Comment by xauthulei** [comment]
> @xauthulei it looks fine but test fails..
> 
> /approve

@qiujian16 Fixed the test error, please review this again, Thanks

File: pkg/cmd/enable/addons/cmd.go [hunk 2]
```diff
@@ -46,6 +46,7 @@ func NewCmd(clusteradmFlags *genericclioptionsclusteradm.ClusteradmFlags, stream
 	}
 
 	cmd.Flags().StringVar(&o.names, "names", "", "Names of the add-on to deploy (comma separated)")
+	cmd.Flags().StringVar(&o.namespace, "ns", "open-cluster-management-agent-addon", "Specified namespace to enable addon")
```
Comment (qiujian16) [inline pkg/cmd/enable/addons/cmd.go:49]: could we do it like kubectl that has `--namespace` and `-n` ?

**Pull request merged**